compute_transfer_examples/collection_transfer_requires_flow/compute_transfer_example_1_definition.json

```
@@ -0,0 +1,122 @@
+{
+  "StartAt": "SetConstants",
+  "States": {
+    "SetConstants": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "gcs_base_path": "/",
+        "compute_endpoint_id": "<INSERT YOUR COMPUTE ENDPOINT ID HERE>",
+        "compute_function_id": "<INSERT YOUR COMPUTE FUNCTION ID HERE>",
+        "compute_output_directory.=": "'/' + _context.run_id + '/'"
+      },
+      "ResultPath": "$.constants",
+      "Next": "GetDestinationPath"
+    },
+    "GetDestinationPath": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "path.=": "getattr('destination_path', transfer_settings.DATA[0].destination_path)"
+      },
+      "ResultPath": "$.destination_path",
+      "Next": "InitIterateSourcePaths"
+    },
+    "InitIterateSourcePaths": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "src_paths_iterator_pos": 0,
+        "src_paths": []
+      },
+      "ResultPath": "$.iterate_source_paths",
+      "Next": "IterateSourcePathsConditional"
+    },
+    "IterateSourcePathsConditional": {
+      "Type": "ExpressionEval",
+      "Parameters": {
+        "conditional.=": "len(transfer_settings.DATA) > iterate_source_paths.src_paths_iterator_pos"
+      },
+      "ResultPath": "$.iterate_source_paths_conditional",
```

Why use `ResultPath` in this case? It seems like it's leading to later `...conditional.conditional`, which I think can be avoided:
```suggestion
        "iterate_source_paths_conditional.=": "len(transfer_settings.DATA) > iterate_source_paths.src_paths_iterator_pos"
      },
```

---

This might be obviated by my feedback below on the loop structure, but I wanted to mention that we don't always have to nest things.

---

Nevermind this -- I forgot that with no `ResultPath`, a flow will end up clobbering data. It feels awkward in terms of naming still, but removing `ResultPath` isn't the right way to fix it. Still, replacing the loop with an expression may be better.

---

I was wondering about this, I could not figure out how to get rid of `ResultPath` for the life of me 😆. In my messing around though, I did discover that you have have the empty string "" as a key if in an ExpressionEval you set a variable to '.='. Not sure if this is known behavior, thought I'd call it out though.